sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -71,7 +71,9 @@ object CurrentOrigin {
   }
 }
 
+// scalastyle:off
```

Comment:
I tried overriding equals but it seems to break things pretty badly because it seemingly overrides the case-class equals which would be auto-generated in subclasses.